src/Sentry/BaggageHeader.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Baggage Header for dynamic sampling.
+    /// </summary>
+    /// <seealso href="https://develop.sentry.dev/sdk/performance/dynamic-sampling-context"/>
+    /// <seealso href="https://www.w3.org/TR/baggage"/>
+    internal class BaggageHeader
+    {
+        internal const string HttpHeaderName = "baggage";
+        internal const string SentryKeyPrefix = "sentry-";
+
+        // https://www.w3.org/TR/baggage/#baggage-string
+        // "Uniqueness of keys between multiple list-members in a baggage-string is not guaranteed."
+        // "The order of duplicate entries SHOULD be preserved when mutating the list."
+
+        public IReadOnlyList<KeyValuePair<string, string>> Members { get; }
+
+        private BaggageHeader(IEnumerable<KeyValuePair<string, string>> members) =>
+            Members = members.ToList().AsReadOnly();
```

Comment:
given Members is internal. and we dont need to care about people bypassing the `IReadOnlyList` contract. we can omit the `.AsReadOnly()`